openmdao/core/driver.py

```diff
@@ -523,6 +540,52 @@ def get_constraint_metadata(self):
         """
         return self._cons
 
+    def _gather_vars(self, vec, varnames):
+        '''
+        Gathers and returns only variables listed in 
+        `varnames` from the vector `vec`
+        '''
+        local_vars = []
+
+        for key, value in iteritems(vec):
+            if key in varnames:
+                if self.root.comm.rank == self.root._owning_ranks[key]:
+                    local_vars.append((key, value['val']))
+
+        all_vars = self.root.comm.gather(local_vars, root=0)
+
+        if self.root.comm.rank == 0:
+            return OrderedDict(itertools.chain(*all_vars))
```

Comment:
Is order really important here?  If not, just using a plain dict would be more efficient.